File: problem_set/one-away-lcci/one-away-lcci.cpp

```
// @Title: 一次编辑 (One Away LCCI)
// @Author: robert.sunq
// @Date: 2020-03-23 23:39:43
// @Runtime: 0 ms
// @Memory: 7.4 MB

class Solution {
public:
    bool oneEditAway(string first, string second) { 
        int len_f = first.length();
        int len_s = second.length();
        if(abs(len_f-len_s) >1) return false;
        int counts = 0;
        if(len_s == len_f){
            for(int i=0;i<len_f;i++){
                if(first[i] != second[i]) counts++;
                if(counts >1) return false;

            }
            return true;
        }
        
        if(len_f >len_s){
            int i=0,j=0;
            counts = 0;
             while(i<len_f && j<len_s){
               if(first[i] != second[j]){
                  counts++;
                  if(counts >1) return false;
                  if(first[i+1] == second[j])
                     i++;
                  else return false; 
            }
            i++;
            j++;
          }  
          return true; 
        }
        else {
             int i=0,j=0;
            counts = 0;
             while(j<len_f && i<len_s){
               if(first[j] != second[i]){
                  counts++;
                  if(counts >1) return false;
                  if(first[j] == second[i+1])
                     i++;
                  else return false; 
            }
             i++;
             j++;
          }
          return true;   
        }
       

    }
};

File: problem_set/one-away-lcci/one-away-lcci.java

```
// @Title: 一次编辑 (One Away LCCI)
// @Author: robert.sunq
// @Date: 2021-08-02 21:28:20
// @Runtime: 2 ms
// @Memory: 38.3 MB

class Solution {
    public boolean oneEditAway(String first, String second) {
        if(Math.abs(first.length()-second.length()) > 1) return false;

        int i = 0,j = 0,count = 0;
        while(i<first.length() && j<second.length()){
            if(first.charAt(i) == second.charAt(j)){
                i++;
                j++;
                continue;
            }
            count++;
            if(count > 1) return false;
            // 第一次不匹配时，两字符串不等时表示长的需要删除一个字符，及长的指针后移一位。
            if(first.length() < second.length()){
                j++;
            } else if(first.length() > second.length()){
                i++;
            } else {
                i++;
                j++;
            }
        }
        return true;
    }
}
